pkg/client/client_venconn.go

```
@@ -0,0 +1,215 @@
+// are referring to as its client-go cache will remain empty.
+//
+// The http.Client is used for Venafi and Vault, not for Kubernetes. The
+// `installNS` is the namespace in which the agent is running in. The passed
+// `restcfg` is not mutated. `trustedCAs` is only used for connecting to Venafi
+// Cloud and Vault and can be left nil.
+func NewVenConnClient(restcfg *rest.Config, agentMetadata *api.AgentMetadata, installNS, venConnName, venConnNS string, trustedCAs *x509.CertPool) (*VenConnClient, error) {
+	// TODO(mael): The rest of the codebase uses the standard "log" package,
+	// venafi-connection-lib uses "go-logr/logr", and client-go uses "klog". We
+	// should standardize on one of them, probably "slog".
+	ctrlruntimelog.SetLogger(logr.Logger{})
+
+	if installNS == "" {
+		return nil, errors.New("programmer mistake: installNS must be provided")
+	}
+	if venConnName == "" {
+		return nil, errors.New("programmer mistake: venConnName must be provided")
+	}
+	if venConnNS == "" {
+		return nil, errors.New("programmer mistake: venConnNS must be provided")
+	}
+
+	restcfg = rest.CopyConfig(restcfg)
+	restcfg.Impersonate = rest.ImpersonationConfig{
+		UserName: fmt.Sprintf("system:serviceaccount:%s:venafi-connection", installNS),
+	}
+
+	// TLS-related configuration such as root CAs and client certs are contained
+	// in the restcfg; let's create an http.Client that uses them.
+	httpCl, err := rest.HTTPClientFor(restcfg)
+	if err != nil {
+		return nil, fmt.Errorf("while turning the REST config into an HTTP client: %w", err)
+	}
+
+	restMapper, err := apiutil.NewDynamicRESTMapper(restcfg, httpCl)
+	if err != nil {
+		return nil, fmt.Errorf("while creating the REST mapper: %w", err)
+	}
+
+	// This Kubernetes client only needs to be able to read and write the
+	// VenafiConnection resources and read Secret resources.
+	scheme := runtime.NewScheme()
+	_ = venapi.AddToScheme(scheme)
+	_ = corev1.AddToScheme(scheme)
```

Why discard these errors?

---

I discarded these errors because the tests would fail if these two schemas hadn't loaded properly. I guess I saved myself a few keystrokes... I don't think this will bite me later